store/src/java/com/zimbra/cs/imap/ImapHandler.java

```diff
@@ -1330,6 +1347,56 @@ boolean doLOGOUT(String tag) throws IOException {
         return false;
     }
 
+    private boolean checkZimbraAuth() {
+        return authenticator != null && authenticator.getMechanism().equals(ZimbraAuthenticator.MECHANISM);
+    }
+
+    boolean doFLUSHCACHE(String tag, CacheEntryType[] types, CacheEntry[] entries) throws IOException {
+        if (!checkState(tag, State.AUTHENTICATED)) {
+            return true;
+        } else if (!checkZimbraAuth()) {
+            sendNO(tag, "must be authenticated with X-ZIMBRA auth mechanism");
+            return true;
+        }
+        for (CacheEntryType type: types) {
+            try {
+                AuthToken authToken = ((ZimbraAuthenticator) authenticator).getAuthToken();
+                AdminAccessControl aac = AdminAccessControl.getAdminAccessControl(authToken);
+                Server localServer = Provisioning.getInstance().getLocalServer();
+                try {
+                    aac.checkRight(localServer, Admin.R_flushCache);
+                } catch (ServiceException e) {
+                    ZimbraLog.imap.error("insufficient rights for flushing cache on IMAP server", e);
+                    return canContinue(e);
+                }
+                Provisioning.getInstance().flushCache(type, entries);
+            } catch (ServiceException e) {
+                ZimbraLog.imap.error("error flushing cache on IMAP server", e);
+                return canContinue(e);
+            }
+        }
+        sendOK(tag, "FLUSHCACHE completed");
+        return true;
+    }
+
+    boolean doRELOADLC(String tag) throws IOException {
+        if (!checkState(tag, State.AUTHENTICATED)) {
+            return true;
+        } else if (!checkZimbraAuth()) {
+            sendNO(tag, "must be authenticated with X-ZIMBRA auth mechanism");
+            return true;
+        }
+        try {
```

Comment:
Yes, as-is, non-admin users would be able to do this. Perhaps I should update doRELOADLC so that only admin users can run this command? That would put it more in line with how ReloadLocalConfigRequest works.